src/haddock/modules/sampling/gdock/__init__.py

```diff
@@ -114,19 +111,32 @@ def _run(self):
         input_toml += f'B = \"{input["B"]}\"' + os.linesep
         input_toml += os.linesep
 
-        with working_directory(self.path):
-            with open('input.toml', 'w') as inp_fh:
-                inp_fh.write(input_toml)
+        # ===============
+        # Placeholder, this is not yet implemented in gdock
+        # input_toml += '[parameters]' + os.linesep
+        # input_toml += 'population_size = 100' + os.linesep
+        # input_toml += 'max_number_of_generations = 50' + os.linesep
+        # ===============
+
+        with open('input.toml', 'w') as inp_fh:
+            inp_fh.write(input_toml)
 
-            cmd = f'{sys.executable} {gdock_exec} --dry input.toml'
+        cmd = f'{sys.executable} {gdock_exec} --dry input.toml'
 
-            subprocess.call(cmd, shell=True)
+        subprocess.call(cmd, shell=True)
 
         # retrieve the structures
         output_structures = []
-        structure_folder = Path(self.path, 'gdock-integration/structures')
+        structure_folder = Path('gdock-integration/structures')
         for model in structure_folder.glob('*pdb'):
-            pdb = PDBFile(model, path=model.parent)
+            # Make sure the output is tidy, this should be handled
+            #  by gdock, but check it here to be sure
+            tmp_f = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
I think is best if you write this with the following:

```python
from pdbtools import pdb_tidy 
with open(model, 'r') as fin:
    lines = list(pdb_tidy.run(fin, strick=False))  # be explicit in the `strict`

with open(model, 'w') as fout:
    fout.write(os.linesep.join(lines))
```
Don't know if the `os.linesep` is needed. With this, you avoid creating a temporary file. I think after #144 this part is no longer needed. I will have that in mind.

Aside from this, :rocket:

Reply:
thanks! I keep trying to find reasons to use tempfile but there is never a good one 😢

Reply:
Yeah, we are always trying to reduce the disk usage, so... :stuck_out_tongue_closed_eyes: